src/aks-preview/azext_aks_preview/_help.py

```diff
@@ -1003,6 +1003,164 @@
+                                          If monitoring addon is enabled --no-wait argument will have no effect
+        virtual-node                    - enable AKS Virtual Node. Requires --subnet-name to provide the name of an existing subnet for the Virtual Node to use.
+        azure-policy                    - enable Azure policy. The Azure Policy add-on for AKS enables at-scale enforcements and safeguards on your clusters in a centralized, consistent manner.
+                                          Learn more at aka.ms/aks/policy.
+        ingress-appgw                   - enable Application Gateway Ingress Controller addon (PREVIEW).
+        open-service-mesh               - enable Open Service Mesh addon (PREVIEW).
+        gitops                          - enable GitOps (PREVIEW).
+        azure-keyvault-secrets-provider - enable Azure Keyvault Secrets Provider addon (PREVIEW).
+parameters:
+  - name: --addon -a
+    type: string
+    short-summary: Specify the Kubernetes addon to enable.
+  - name: --workspace-resource-id
+    type: string
+    short-summary: The resource ID of an existing Log Analytics Workspace to use for storing monitoring data.
+  - name: --enable-msi-auth-for-monitoring
+    type: bool
+    short-summary: Send monitoring data to Log Analytics using the cluster's assigned identity (instead of the Log Analytics Workspace's shared key).
+  - name: --subnet-name -s
+    type: string
+    short-summary: The subnet name for the virtual node to use.
+  - name: --appgw-name
+    type: string
+    short-summary: Name of the application gateway to create/use in the node resource group. Use with ingress-azure addon.
+  - name: --appgw-subnet-prefix
+    type: string
+    short-summary: Subnet Prefix to use for a new subnet created to deploy the Application Gateway. Use with ingress-azure addon.
+  - name: --appgw-subnet-cidr
+    type: string
+    short-summary: Subnet CIDR to use for a new subnet created to deploy the Application Gateway. Use with ingress-azure addon.
+  - name: --appgw-id
+    type: string
+    short-summary: Resource Id of an existing Application Gateway to use with AGIC. Use with ingress-azure addon.
+  - name: --appgw-subnet-id
+    type: string
+    short-summary: Resource Id of an existing Subnet used to deploy the Application Gateway. Use with ingress-azure addon.
+  - name: --appgw-watch-namespace
+    type: string
+    short-summary: Specify the namespace, which AGIC should watch. This could be a single string value, or a comma-separated list of namespaces. Use with ingress-azure addon.
+  - name: --enable-sgxquotehelper
+    type: bool
+    short-summary: Enable SGX quote helper for confcom addon.
+  - name: --enable-secret-rotation
+    type: bool
+    short-summary: Enable secret rotation. Use with azure-keyvault-secrets-provider addon.
```

Comment:
Could please add some test cases to cover all supported options for enable/update?

Reply:
I can but the base code behind each of these options is the same as the code behind `enable-addons`, which would lead to a lot of duplicated tests that are checking for the exact same thing. Do you think it is still worth it since the new code I wrote is not related to these options.

Reply:
I think this should be necessary. Probably these options are passed and handled correctly for now, but future changes may introduce errors. There is no need to create a test case for each option. Options belonging to the same addon can be placed in the same test case for testing. Don't forget to test both `enable` and `update`:wink:

Reply:
Hi Fuming, I was wondering if there is a doc that notes what each parameter does so I can test it appropriately. I realized that there actually weren't test cases for each of the options with `enable-addons`, so I'll have to make new tests.

Reply:
I think adding new test cases for addon parameters is a little bit out of the scope of this pr because addon owner has more knowledge to do this. But I think we can reuse the existing unit test cases for enable/disable commands. For example, we can add extra lines to test `addon disable/enable` commands to the end of the `test_aks_byo_appgw_with_ingress_appgw_addon` test case.

Reply:
There're already >20 unit test cases for various addons if you search `addonProfiles` in the test_aks_commands.py